doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,255 @@
+==================================
+Experiment Tracking with Ray Train
+==================================
 
-Most experiment tracking libraries work out-of-the box with Ray Train. This means
-you can use the regular tracking library APIs to report your results, models, and
-other artifacts.
+.. note::
+    This guide is relevant for all DataParallelTrainer, including TorchTrainer and TensorflowTrainer.
 
-Ray Train also exposes logging callbacks that automate some of these tasks.
+Most experiment tracking libraries work out-of-the-box with Ray Train. 
+In the following guide, you will learn how to use your favorite experiment tracking libraries 
+while doing distributed data parallel training with Ray Train. 
 
-.. _train-experiment-tracking-native:
+The guide will be focusing on TorchTrainer although the ideas should apply to DataParallelTrainer
+in general.
 
-Using native experiment tracking libraries
-------------------------------------------
+Ray Train lets you use native experiment tracking libraries by customizing your tracking 
+logic inside ``train_loop_per_worker``. 
+This gives you the highest degree of freedom to decide when and where to log params, metrics, 
+models or even media contents. 
+This also allows you to use some native integrations that these tracking frameworks have with 
+specific training frameworks, for example ``mlflow.pytorch.autolog()``, 
+``lightning.pytorch.loggers.MLFlowLogger`` etc. 
+In this way, you get to port your experiment tracking logic to Ray Train with minimal change. 
 
-.. tab-set::
+Mental picture of experiment tracking in Ray Train
+==================================================
 
-    .. tab-item:: PyTorch
+There are many experiment tracking frameworks out there. So instead of trying to 
+cover every of them with a variety of usages, let’s first look at how to think 
+about experiment tracking with Ray Train DataParallelTrainer, and how it may be 
+different than running native training code in a single process. This mental 
+picture helps you reason about some errors and makes debugging a lot easier.
 
-        You can use experiment tracking libraries such as Weights & Biases, Mlflow, or
-        Comet directly in your Ray Train training loop.
+In Ray Train DataParallelTrainer, interaction with experiment tracking services 
+happens inside of ``train_loop_per_worker``. ``train_loop_per_worker`` is run as a separate process on 
+an arbitrary node. This is different from single process training in one 
+node in the following ways.
 
-        There are two things to keep in mind:
+``train_loop_per_worker`` needs to have access to the tracking service
+----------------------------------------------------------------------
 
-        1. Your code is executed in parallel on many workers. However, you often only want to report
-        results from one of these workers (usually the first worker - the "rank 0" worker).
+Usually this is trivial for online externally hosted tracking services. 
```

Comment:
How many exp tracking tools are we going to discuss in this user guide? Wandb, MLFlow, Tensorboard? 

Do you think we need to ensure all sections have info for each framework? (We are missing wandb in this section.) And we can separate all by tabs?

Reply:
wandb is trivial for this section, which says a lot about its simplicity actually.
mlflow is difficult in the sense it has 4/5 flavors? It took me some time to set it up myself and I imagine the same for our users. Hence this section.